docs/source/dev_guide/testing.rst

```diff
@@ -1,22 +1,60 @@
 Testing E3SM Diagnostics
 ==============================================
 
-Unit tests
-----------
+Unit and integration tests
+--------------------------
 
-Run all unit tests by doing the following:
+Run all automated tests by doing the following:
 
     .. code::
 
         pip install . # Install your changes
-        ./tests/test.sh # Run all unit tests
+        ./tests/test.sh # Run all unit and integration tests
+
+
+If these tests pass, you're done. If they fail unexpectedly however,
+your code might have changed what the output looks like.
+
+``ls tests/system/image_check_failures`` will show your all the images that differ from expected.
+
+If you see unexpected images listed, you'll have to investigate why your
+code changed what the images look like.
+
+If the only images listed are ones you expected to see changed,
+then you need to update the expected images.
+
+
+    .. code::
+
+        cd /lcrc/group/e3sm/public_html/e3sm_diags_test_data/integration/expected
+        cat README.md
+        # This will show you the version, date, and hash of the current expected images.
+        # Using that information do the following:
+        mv integration_test_images previous_output/integration_test_images_<version>_<date>_<hash>
+        # `cd` back into your E3SM Diags directory.
+        # Your output will now become the new expectation.
+        mv all_sets_results /lcrc/group/e3sm/public_html/e3sm_diags_test_data/integration/expected/integration_test_images
+        cd /lcrc/group/e3sm/public_html/e3sm_diags_test_data/integration/expected
+
+Now, edit ``README.md``. The version should be the version of E3SM Diags you ran
+``./tests/test.sh`` with,
+the date should be the date you ran ``./tests/test.sh`` on,
+and the hash should be for the top commit shown by ``git log``.
+
+    .. warning::
+
+        If you later update your commit history before merging into ``master``, then
+        change the hash to be the latest commit in your pull request right before
+        merging.
```

Comment:
Perhaps I should note to do this step AFTER merging, so that `<hash>` will match the hash of the squashed merge commit (which there is no way to know ahead of time).

Reply:
And speaking of merge methods, should we prefer "Squash and merge" or "Rebase and merge"? I've been doing "Create a merge commit".

Reply:
> Perhaps I should note to do this step AFTER merging, so that `<hash>` will match the hash of the squashed merge commit (which there is no way to know ahead of time).

I agree, the hash changes when squashing so it needs to match.

> And speaking of merge methods, should we prefer "Squash and merge" or "Rebase and merge"? I've been doing "Create a merge commit".

Jill and I have been using "Squash and merge", which allows us to avoid pushing a merge commit along with our squashed commit.

Reply:
> Jill and I have been using "Squash and merge", which allows us to avoid pushing a merge commit along with our squashed commit.

What's the advantage of that over "Rebase and merge" though?